src/box/vy_log.c

```diff
@@ -1057,11 +1056,36 @@ vy_log_begin_recovery(const struct vclock *vclock, bool force_recovery)
 		/*
 		 * Last vy_log log is newer than the last snapshot.
 		 * This can't normally happen, as vy_log is rotated
-		 * after snapshot is created. Looks like somebody
+		 * in a short gap between checkpoint wait and commit.
+		 * However, if memtx for some reason fails to commit its
+		 * changes, instance will crash leaving .inprogress snap
+		 * and corresponding (already rotated) vylog.
+		 * Another and simpler reason is the case when somebody
 		 * deleted snap file, but forgot to delete vy_log.
+		 * So in case we are anyway in force recovery mode, let's
+		 * try to delete last .vylog file and continue recovery process.
 		 */
-		diag_set(ClientError, ER_MISSING_SNAPSHOT);
-		return NULL;
+		if (! force_recovery) {
+			diag_set(ClientError, ER_MISSING_SNAPSHOT);
+			say_verbose("To bootstrap instance try to remove last "
```

Comment:
Would be better to make it at least `info` or `warning`. `Verbose` might not be turned on. The same below.

Reply:
Ok, let's use info level